test/index.type-test.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { Text } from 'react-native'
+import styled from '../src/index'
+
+const StyledText = styled(Text)(({ transparent }: { transparent?: boolean; color: string }) => ({ flex: 1, opacity: transparent ? 0.5 : 1 }))
+const StyledScalableText = StyledText.extend(({ big }: { big?: boolean }) => ({ fontSize: big ? 20 : 10 }))
+const StyledTextWithObjectProps = styled(Text)({ flex: 1, opacity: 1 })
+
+const MyScreen = () => {
+  return (
+    <>
+      <StyledText color="red" transparent>Hello Transparent World</StyledText>
+      {/* should also work without transparent prop */}
+      <StyledText color="red">Hello World</StyledText>
+      {/* @ts-expect-error -- should not work without required color prop */}
+      <StyledText>Hello World</StyledText>
+      {/* @ts-expect-error -- big is only available on StyledScalableText */}
+      <StyledText color="red" big>Hello World</StyledText>
+      <StyledScalableText color="red" transparent big>Hi</StyledScalableText>
+      {/* @ts-expect-error -- transparent prop is not allowed */}
+      <StyledTextWithObjectProps transparent>Hi</StyledTextWithObjectProps>
```

Comment:
note these type tests, they are a useful development tool but also help to prevent regressions

Reply:
how we avoid passing transaprent to actual native component?

Reply:
ah didn't notice comment, all good

Reply:
this is just typing with no runtime effects and supports the previous usage patterns that we have
> The reason is that with shakl with pass this custom prop down to component - this is not expected. right now react-native allows that

it doesn't change anything about that